presctl/cmd/entities_test.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"testing"
+
+	"github.com/Microsoft/presidio/presctl/cmd/entities"
+)
+
+// clientMock is a mock for http client
+type clientMock struct {
+	returningCode int
+}
+
+// TestIsJson tests that the method differentiate between a valid to a non valid json string
+func TestIsJson(t *testing.T) {
+	nonJSONStr := "A simple string"
+	if isJSON(nonJSONStr) {
+		t.Fatal("isJson failed. the given string is not a valid JSON, yet the method returned true")
+	}
+
+	nonJSONStr = `{
+	                itemA: "A"
+                  }`
+	if isJSON(nonJSONStr) {
+		t.Fatal("isJson failed. the given string is not a valid JSON, yet the method returned true")
+	}
+
+	validJSONStr := `{
+	            	  "itemA": "A"
+                     }`
+	if !isJSON(validJSONStr) {
+		t.Fatal("isJson failed. the given is a valid JSON, yet the method returned false")
+	}
+}
+
+// TestFailingCreateTemplate tests that when presidio returns an invalid status code the cli exists with an error
+func TestFailingCreateTemplate(t *testing.T) {
+	// we expect the exit code to be 1, if the returned status code is not as expected (not 201)
+	if os.Getenv("BE_CRASHER") == "1" {
+		mockFail := &clientMock{returningCode: 400}
+		entities.CreateTemplate(mockFail, "myproj", "anonymize", "template1", "template text")
+		return
+	}
+	cmd := exec.Command(os.Args[0], "-test.run=TestFailingCreateTemplate")
```

Comment:
`os.Args[0]` is an external dependency. 
The use of env. variable and app arguments is not clear to me in this test.

Reply:
see above comment